pkg/virtual/options/options.go

```diff
@@ -92,22 +90,17 @@ func (o *Options) NewVirtualWorkspaces(
 		return nil, err
 	}
 
-	all, err := merge(workspaces, syncer, apiexport, initializingworkspaces)
+	all, err := merge(workspaces, syncer, apiexports, initializingworkspaces)
 	if err != nil {
 		return nil, err
 	}
 	return all, nil
 }
 
-func merge(sets ...map[string]framework.VirtualWorkspace) (map[string]framework.VirtualWorkspace, error) {
-	workspaces := make(map[string]framework.VirtualWorkspace)
+func merge(sets ...[]rootapiserver.NamedVirtualWorkspace) ([]rootapiserver.NamedVirtualWorkspace, error) {
+	var workspaces []rootapiserver.NamedVirtualWorkspace
 	for _, set := range sets {
-		for name, w := range set {
-			if _, ok := workspaces[name]; ok {
-				return nil, fmt.Errorf("duplicate virtual workspace %q", name)
-			}
-			workspaces[name] = w
-		}
+		workspaces = append(workspaces, set...)
```

Comment:
no need to check for duplicates anymore?

Reply:
good point. Readding.

Reply:
done